src/editor.ts

```diff
@@ -91,55 +90,26 @@ export class Editor {
 	}
 
 	// Kill to end of line
-	async kill(): Promise<boolean> {
+	kill(): void {
 		// Ignore whatever we have selected before
-		await vscode.commands.executeCommand("emacs.exitMarkMode")
-
-		let startPos = this.getCurrentPos(),
-			isOnLastLine = Editor.isOnLastLine()
-
-		// Move down an entire line (not just the wrapped part), and to the beginning.
-		await vscode.commands.executeCommand("cursorMove", { to: "down", by: "line", select: false })
-		if (!isOnLastLine) {
-			await vscode.commands.executeCommand("cursorMove", { to: "wrappedLineStart" })
-		}
-
-		let endPos = this.getCurrentPos(),
-			range = new vscode.Range(startPos, endPos),
-			txt = vscode.window.activeTextEditor.document.getText(range)
-
-		// If there is something other than whitespace in the selection, we do not cut the EOL too
-		if (!isOnLastLine && !txt.match(/^\s*$/)) {
-			await vscode.commands.executeCommand("cursorMove", {to: "left", by: "character"})
-			endPos = this.getCurrentPos()
-		}
-
-		// Select it now, cut the selection, remember the position in case of multiple cuts from same spot
-		this.setSelection(startPos, endPos)
-		let promise = this.cut(this.lastKill != null && startPos.isEqual(this.lastKill))
-
-		promise.then(() => {
-			this.justDidKill = true
-			this.lastKill = startPos
+		vscode.commands.executeCommand("emacs.exitMarkMode")
+		vscode.commands.executeCommand("cursorMove", {
+			to:"wrappedLineEnd",
+			select:true
 		})
-
-		return promise
+		// editor.action.clipboardCutAction is too slow!
+		vscode.commands.executeCommand("editor.action.clipboardCopyAction");
+		vscode.commands.executeCommand("deleteAllRight");
 	}
 
 	copy(): void {
 		vscode.commands.executeCommand("editor.action.clipboardCopyAction")
 		vscode.commands.executeCommand("emacs.exitMarkMode")
 	}
 
-	cut(appendClipboard?: boolean): Thenable<boolean> {
-		if (appendClipboard) {
```

Comment:
The append option was used because successive C-k cut operations from the same location append to the clipboard instead of just replacing it.